AmplifyPlugins/API/AWSAPICategoryPluginIntegrationTests/GraphQL/GraphQLWithUserPoolIntegrationTests/README.md

```diff
@@ -0,0 +1,49 @@
+## GraphQL with UserPool Auth Integration Tests
+
+The following steps show how to set up an GraphQL endpoint with AppSync. The auth configured will be Cognito UserPools. This set up is used to run the tests in `GraphQLWithUserPoolIntegrationTests.swift`.
+
+### Set-up
+
+1. `amplify init`
+
+2. `amplify add api`
+
+```perl
+? Please select from one of the below mentioned services: GraphQL
+? Provide API name: `<APIName>`
+? Choose the default authorization type for the API `Amazon Cognito User Pool`
+? Do you want to use the default authentication and security configuration? `Default configuration`
+? How do you want users to be able to sign in? `Email`
+? Do you want to configure advanced settings? `No, I am done.`
+? Do you want to configure advanced settings for the GraphQL API `No, I am done.`
+? Do you have an annotated GraphQL schema? `No`
+? Do you want a guided schema creation? `Yes`
+? What best describes your project: `Single object with fields (e.g., “Todo” with ID, name, description)`
+? Do you want to edit the schema now? `No`
+```
+
+The guided schema provided should look like this: 
+```json
+type Todo @model {
```

Comment:
Why are we using a TODO model for this test and a Post/comment model for the other ones?

Reply:
the todo model tests were written first, against the document based GraphQL APIs and with the userpool auth tested. the post/comment model for the other one is after we introduced the Model based GraphQL APIs with a shared model in AmplifyTestCommon. So reason being was just because of when the test was needed at the time. in the future, I would like to have a test that uses @auth directives with userpools and Model based. Currently model based is using API Key